acceptance/tests/ticket_1123_facter_with_invalid_locale.rb

```diff
@@ -1,15 +1,23 @@
-test_name 'ticket 1123  facter should not crash with invalid locale setting'
-confine :except, :platform => 'windows'
-confine :except, :platform => /^cisco_/ # See CISCO-43
-confine :except, :platform => /^huawei/ # See HUAWEI-24
+# ticket FACT-1123
+test_name 'C89524: facter should not crash with invalid locale setting' do
+  tag 'risk:high'
 
-agents.each do |host|
-  step 'set an invalid value for locale and run facter'
-  result = on host, 'LANG=ABCD facter facterversion'
-  if host['platform'] !~ /solaris|aix|cumulus|osx/
-    fail_test 'facter did not warn about the locale' unless
-      result.stderr.include? 'locale environment variables were bad; continuing with LANG=C LC_ALL=C'
+  confine :except, :platform => 'windows'
+  confine :except, :platform => /^cisco_/ # See CISCO-43
+  confine :except, :platform => /^huawei/ # See HUAWEI-24
+
+  agents.each do |agent|
+    step 'facter should run when started with an invalid locale' do
+      on(agent, facter('facterversion'), :environment => {'LANG' => 'ABCD'}) do |facter_output|
+        assert_match(/^\d+\.\d+\.\d+$/, facter_output.stdout, 'facter did not continue running')
+
+        if agent['platform'] !~ /solaris|aix|cumulus|osx/
```

Comment:
Because line 12 makes sure that facter ran successfully and returned a value rather than crashing. So even on the operating system's that don't print the error we at least ensure that facter doesn't crash.